internal/config/registry.go

```diff
@@ -42,8 +42,8 @@ func (cfg *registry) parseConfigValues() {
 	}
 }
 
-func hasNonEmptyCredentials(authority, password, token string) bool {
-	return authority != "" && password != "" || authority != "" && token != ""
+func hasNonEmptyCredentials(authority, username, password, token string) bool {
+	return authority != "" && password != "" && username != "" || authority != "" && token != ""
```

Comment:
this is great, I didn't realize that by spotting some extra test cases it would surface this as a needed change, thanks for addressing it